File: languages/python/bloock/entity/identity/credential_verification.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class CredentialVerification:
    def __init__(self, timestamp: int, issuer: str, revocation: int) -> None:
        self.timestamp = timestamp
        self.issuer = issuer
        self.revocation = revocation

    @staticmethod
    def from_proto(c: proto.CredentialVerification) -> CredentialVerification:
        return CredentialVerification(
            timestamp=c.timestamp,
            issuer=c.issuer,
            revocation=c.revocation
        )

    def to_proto(self) -> proto.CredentialVerification:
        return proto.CredentialVerification(
            timestamp=self.timestamp,
            issuer=self.issuer,
            revocation=self.revocation,
        )

File: languages/php/Bloock/Entity/Identity/CredentialVerification.php
<?php

namespace Bloock\Entity\Identity;

class CredentialVerification
{
    private int $timestamp;
    private string $issuer;
    private int $revocation;

    /**
     * @param int $timestamp
     * @param string $issuer
     * @param int $revocation
     */
    public function __construct(int $timestamp, string $issuer, int $revocation)
    {
        $this->timestamp = $timestamp;
        $this->issuer = $issuer;
        $this->revocation = $revocation;
    }

    public static function fromProto(\Bloock\CredentialVerification $res): CredentialVerification
    {
        return new CredentialVerification($res->getTimestamp(), $res->getIssuer(), $res->getRevocation());
    }

    public function toProto(): \Bloock\CredentialVerification
    {
        $p = new \Bloock\CredentialVerification();
        $p->setTimestamp($this->timestamp);
        $p->setIssuer($this->issuer);
        $p->setRevocation($this->revocation);
        return $p;
    }
}